front/src/common/Map/Layers/InfraObjectLayers/InfraObjectLayers.tsx

```diff
@@ -18,84 +19,97 @@ type InfraObjectLayersProps = {
       layerOrder={LAYER_GROUPS_ORDER[LAYERS.TRACKS.GROUP]}
       infraID={infraId}
     />
-
-    <Routes
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.ROUTES.GROUP]}
-      infraID={infraId}
-    />
-
-    <OperationalPoints
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.OPERATIONAL_POINTS.GROUP]}
-      operationnalPointId={hoveredOperationalPointId}
-      infraID={infraId}
-    />
-
-    <Electrifications
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.ELECTRIFICATIONS.GROUP]}
-      infraID={infraId}
-    />
-
-    <NeutralSections
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.DEAD_SECTIONS.GROUP]}
-      infraID={infraId}
-    />
-
-    <BufferStops
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.BUFFER_STOPS.GROUP]}
-      infraID={infraId}
-    />
-
-    <Detectors
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.DETECTORS.GROUP]}
-      infraID={infraId}
-    />
-
-    <Switches
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.SWITCHES.GROUP]}
-      infraID={infraId}
-    />
-
-    <SpeedLimits
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.SPEED_LIMITS.GROUP]}
-      punctualLayerOrder={LAYER_GROUPS_ORDER[LAYERS.SPEED_LIMITS_PUNCTUAL.GROUP]}
-      infraID={infraId}
-    />
-
-    <SNCF_PSL
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.SPEED_LIMITS.GROUP]}
-      punctualLayerOrder={LAYER_GROUPS_ORDER[LAYERS.SPEED_LIMITS_PUNCTUAL.GROUP]}
-      infraID={infraId}
-    />
-
-    <Signals
-      sourceTable="signals"
-      colors={colors[mapStyle]}
-      layerOrder={LAYER_GROUPS_ORDER[LAYERS.SIGNALS.GROUP]}
-      infraID={infraId}
-    />
+    {layersSettings.routes && (
+      <Routes
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.ROUTES.GROUP]}
+        infraID={infraId}
+      />
+    )}
+    {layersSettings.operational_points && (
+      <OperationalPoints
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.OPERATIONAL_POINTS.GROUP]}
+        operationnalPointId={hoveredOperationalPointId}
+        infraID={infraId}
+      />
+    )}
+    {layersSettings.electrifications && (
+      <Electrifications
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.ELECTRIFICATIONS.GROUP]}
+        infraID={infraId}
+      />
+    )}
+    {layersSettings.neutral_sections && (
+      <NeutralSections
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.DEAD_SECTIONS.GROUP]}
+        infraID={infraId}
+      />
+    )}{' '}
+    {layersSettings.buffer_stops && (
+      <BufferStops
+        colors={colors[mapStyle]}
+        layerOrder={LAYER_GROUPS_ORDER[LAYERS.BUFFER_STOPS.GROUP]}
+        infraID={infraId}
+      />
+    )}{' '}
```

Comment:
Hm why do you have `{' '}` ? Is it necessary ? 
```suggestion
    )}
```

(nit: you can introduce a blank line between each layer if you want)

Reply:
I'm not sure how it got here, I think my linter might have add it or something. Removing it :)